### ISSUE TITLE
False warning

### ISSUE THREAD

**Issue by mrnolte** [knowsys/rulewerk#173]
Hey there,
I build the current master branch and, when reasoning, in some cases I receive an irritating warning:
`WARN org.semanticweb.rulewerk.core.reasoner.implementation.VLogReasoner - No data statements (facts or datasource declarations) have been provided`. This warning occurres even though I gave some facts. Consider the follwing example:
```java
final var isCausedBy = Expressions.makePredicate("causes", 2);
final var causes = Expressions.makePredicate("isCausedBy", 2);
final var kB = new KnowledgeBase();
final var constant = Expressions.makeAbstractConstant("*");
// causes(*,*)
kB.addStatement(Expressions.makeFact(causes, constant, constant));
// isCausedBy(*,*)
kB.addStatement(Expressions.makeFact(isCausedBy, constant, constant));
final var first1 = Expressions.makeAbstractConstant("first_1");
final var first2 = Expressions.makeAbstractConstant("first_2");
final var second1 = Expressions.makeAbstractConstant("second_1");
final var second2 = Expressions.makeAbstractConstant("second_2");
// isCausedBy(first_1, second_1) :- causes(second_1, first_1)
kB.addStatement(Expressions.makeRule(Expressions.makePositiveLiteral(isCausedBy, first1, second1),
Expressions.makePositiveLiteral(causes, second1, first1)));
// causes(first_2, second_2) :- isCausedBy(second_2, first_2)
kB.addStatement(Expressions.makeRule(Expressions.makePositiveLiteral(causes, first2, second2),
Expressions.makePositiveLiteral(isCausedBy, second2, first2)));
try (final Reasoner reasoner = new VLogReasoner(kB)) {
	if (!reasoner.reason()) {
		throw new RuntimeException();
	}
}
```

Cheers,
Robin
